terraform-tests/cosmosdb_mongo_test.go

```diff
@@ -132,7 +134,19 @@ var _ = Describe("CosmosDB Mongo", Label("cosmosdb-mongo-terraform"), Ordered, f
 					"database_name":       Equal(dbName),
 					"default_ttl_seconds": BeNumerically("==", 777),
 					"shard_key":           Equal("key1"),
-				}))
+
+					"index": BeEquivalentTo([]any{
```

Comment:
Alternatively:
```go
  "index": ConsistOf(
    map[string]any...,
    map[string]any...,
  ),
```

Or:
```go
 "index": ConsistOf(
    MatchAllKeys(
     "keys": ConsistOf("_id"),
     "unique": BeTrue(),
    ),
   ....,
  ),
```

It's debatable which of these is more readable and gives a better message on failure.

Reply:
Not being an expert myself in Gomega Matchers,
I tried different combinations of Matchers without success.
`BeEquivalentTo` helped me get the test passing.

I acknowledge, BeEquivalentTo is probably less useful for:
- Large objects (since I guess it will print the whole object if something fails)
- Deep nesting (same as above)
- When more flexibility is needed (it doesn't allow ignoring missing or additional keys)

I believe It has one very big advantage though, when used appropriately:
- It reads exactly as any other golang struct, without requiring any Gomega knowledge.

This is exactly what helped me debug what types I needed exactly.